src/planner.c

```diff
@@ -417,14 +417,30 @@ classify_relation(const PlannerInfo *root, const RelOptInfo *rel, Hypertable **p
 		case RELOPT_OTHER_MEMBER_REL:
 			rte = planner_rt_fetch(rel->relid, root);
 			parent_rte = get_parent_rte(root, rel->relid);
-			ht = get_hypertable(parent_rte->relid, CACHE_FLAG_CHECK);
 
-			if (NULL != ht)
+			/*
+			 * An entry of reloptkind RELOPT_OTHER_MEMBER_REL might still
+			 * be a hypertable here if it was pulled up from a subquery
+			 * as happens with UNION ALL for example. So we have to
+			 * check for that to properly detect that pattern.
+			 */
+			if (parent_rte->rtekind == RTE_SUBQUERY)
+			{
+				ht = get_hypertable(rte->relid, CACHE_FLAG_CHECK);
+				if (ht != NULL)
+					reltype = TS_REL_HYPERTABLE;
```

Comment:
This is probably the correct classification. But could this be classified as `TS_REL_HYPERTABLE_CHILD` given that the hypertable is child of the subquery? Or should we even another classification for this case?

Reply:
TS_REL_HYPERTABLE_CHILD is the result of hypertable expansion done by postgres because it includes the base table as expanded child (we used to do the same some time ago but haven't for a while). For TS_REL_HYPERTABLE_CHILD no further action is needed while TS_REL_HYPERTABLE still needs expansion which is the case for hypertable pulled up from subqueries. So I think TS_REL_HYPERTABLE is correct, we could potentially add a new classification but I'm not sure what we would gain from it, it would certainly make this patch much bigger.